reana/reana_dev/git.py

```diff
@@ -807,21 +830,48 @@ def git_log(component, number, all):  # noqa: D301
                                to include several runable REANA demo examples;
                          * (7) special value 'ALL' that will expand to include
                                all REANA repositories.
-    :param number: The number of commits to output. [6]
-    :param all: Show all references? [6]
+    :param exclude_components: List of components to exclude from the build.
+    :param number: The number of commits to output. [10]
+    :param graph: Show log graph?
+    :param oneline: Show one-line format?
+    :param patch: Show diff patch?
+    :param all: Show all references?
+    :param no_pager: Avoid output pager?
+    :type range: str
     :type component: str
+    :type exclude_components: str
     :type number: int
+    :type graph: bool
+    :type oneline: bool
+    :type stat: bool
+    :type patch: bool
     :type all: bool
+    :type no_pager: bool
     """
-    for component in select_components(component):
-        cmd = (
-            "git log -n {0} --graph --decorate"
+    if exclude_components:
+        exclude_components = exclude_components.split(",")
+    components = select_components(component, exclude_components)
+    for component in components:
+        cmd = "git log"
+        if no_pager:
+            cmd = "git --no-pager log"
+        if number:
+            cmd += " -n {}".format(number)
+        if graph:
+            cmd += " --graph --decorate"
             ' --pretty=format:"%C(blue)%d%Creset'
             " %C(yellow)%h%Creset %s, %C(bold green)%an%Creset,"
-            ' %C(green)%cd%Creset" --date=relative'.format(number)
-        )
+            ' %C(green)%cd%Creset" --date=relative'
```

Comment:
```python
            cmd += (
                " --graph --decorate"
                ' --pretty=format:"%C(blue)%d%Creset'
                " %C(yellow)%h%Creset %s, %C(bold green)%an%Creset,"
                ' %C(green)%cd%Creset" --date=relative'
            )
```

Reply:
Hmm, interesting that black did not reformat this?!